modules/json/src/smithy4s/json/internals/Cursor.scala

```diff
@@ -65,7 +65,12 @@ private[internals] class Cursor private () {
     top += 1
   }
 
-  def pop(): Unit = top -= 1
+  def pop(): Unit = {
+    if (top > 0) {
+      top -= 1
+    }
+    labelStack(top) = null
```

Comment:
it's suspicious to me that you wouldn't apply the same to the indexStack.

Reply:
Well you can, but don't strictly need to because the function for getting the items out of the stack looks at labelStack(top) to check if it has a value before continuing on to pull the value from the index stack if it doesn't. So the issue came from the labelStack having a value, where it should have been deferring to the index stack

Reply:
gotcha 👍